src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
A cancellation token is an ephemeral object used during the processing of an operation.  This is the pattern .net uses throughout its APIs.  I don't like thinking about it as a singleton.   Esp as different operations may use different tokens, and each token may or may not be canceled.